src/System.Private.Reflection.Execution/src/Internal/Reflection/Execution/ExecutionEnvironmentImplementation.MappingTables.cs

```diff
@@ -127,8 +127,17 @@ public unsafe sealed override bool IsReflectionBlocked(RuntimeTypeHandle runtime
             runtimeTypeHandle = GetTypeDefinition(runtimeTypeHandle);
 
             var moduleHandle = RuntimeAugments.GetModuleFromTypeHandle(runtimeTypeHandle);
-            NativeFormatModuleInfo module = ModuleList.Instance.GetModuleInfoByHandle(moduleHandle);
+            ModuleInfo untypedModuleInfo;
+            NativeFormatModuleInfo module = null;
 
+            //make sure the module is actually NativeFormatModuleInfo, in platforms where
+            //real reflection isnt enabled the module will not be a NativeFormatModuleInfo
+            if (ModuleList.Instance.TryGetModuleInfoByHandle(moduleHandle, out untypedModuleInfo))
+                module = untypedModuleInfo as NativeFormatModuleInfo;
+
+            if(module == null)
+                return true;
```

Comment:
Nit: use C# 7's concise syntax

```c#
if (ModuleList.Instance.TryGetModuleInfoByHandle(moduleHandle, out var untypedModuleInfo)
{
    if(untypedModuleInfo is NativeFormatModuleInfo module)
    {
        return true;
    }
}
```

Reply:
I don't think the code you included does the same job. We still need to get the NativeFormatModuleInfo instance into module, preferably with as few casts as possible. That's why I went with the 'as NativeFormatModuleInfo' rather than 'is NativeFormatModuleInfo'

Reply:
I'll second the C# 7 recommendation minus the use of `var` - our guidelines are `var` only when "obvious", which I take as "When I'm reading the code outside of the VS IDE, I don't have to look beyond that same line to find a reference to the actual type."